services/distribution/src/main/java/app/coronawarn/server/services/distribution/statistics/local/LocalStatisticsToProtobufMapping.java

```diff
@@ -0,0 +1,233 @@
+   */
+  @Bean
+  public Map<Integer, LocalStatistics> constructProtobufLocalStatistics() {
+    Map<Integer, LocalStatistics> localStatisticsMap = new HashMap<>();
+
+    try {
+      Optional<JsonFile> file = this.getFile();
+
+      if (file.isEmpty()) {
+        logger.warn("Stats file is already updated to the latest version. Skipping generation.");
+        return Collections.emptyMap();
+      } else {
+        this.updateETag(file.get().getETag());
+
+        List<LocalStatisticsJsonStringObject> onePerProvinceStatistics = deserializeAndValidate(file);
+
+        onePerProvinceStatistics.forEach(localStatisticsJsonStringObject -> {
+          if (localStatisticsJsonStringObject.getProvinceCode() != null) {
+            int provinceCode = Integer.parseInt(localStatisticsJsonStringObject.getProvinceCode());
+
+            if (isFederalState(localStatisticsJsonStringObject.getProvinceCode())) {
+              fillLocalStatisticsFederalStatesGroupMap(
+                  localStatisticsMap,
+                  provinceCode,
+                  localStatisticsJsonStringObject,
+                  federalStateSupplier(provinceCode, localStatisticsJsonStringObject),
+                  federalStateEnhancer(provinceCode, localStatisticsJsonStringObject)
+              );
+            } else {
+
+              Optional<String> federalStateCodeOptional = findFederalStateByProvinceCode(
+                  localStatisticsJsonStringObject.getProvinceCode());
+
+              federalStateCodeOptional.ifPresent(code -> {
+                fillLocalStatisticsFederalStatesGroupMap(
+                    localStatisticsMap,
+                    Integer.parseInt(federalStateCodeOptional.get()),
+                    localStatisticsJsonStringObject,
+                    administrativeUnitSupplier(provinceCode, localStatisticsJsonStringObject),
+                    administrativeUnitEnhancer(provinceCode, localStatisticsJsonStringObject)
+                );
+              });
+            }
+          }
+        });
+      }
+    } catch (BucketNotFoundException | ConnectionException | FilePathNotFoundException | IOException ex) {
+      logger.error("Local statistics file not generated!", ex);
+    }
+
+    return localStatisticsMap;
+  }
+
+  private Optional<JsonFile> getFile() {
+    var mostRecent = this.localStatisticsDownloadService.getMostRecentDownload();
+    if (mostRecent.isPresent()) {
+      return this.jsonFileLoader.getFileIfUpdated(StatisticType.LOCAL, mostRecent.get().getEtag());
+    } else {
+      return Optional.of(this.jsonFileLoader.getFile(StatisticType.LOCAL));
+    }
+  }
+
+  private void updateETag(String newETag) {
+    var currentTimestamp = TimeUtils.getCurrentUtcHour().toEpochSecond(ZoneOffset.UTC);
+    this.localStatisticsDownloadService.store(currentTimestamp, newETag);
+  }
+
+  /**
+   * Converts an {@link LocalStatisticsJsonStringObject} into a Local Statistics object and handle it addition
```

Comment:
Since the parameter is not used, i do not understand the comment. Maybe you could explain it to me :)

Reply:
What a good remark. I adapted the javadoc and removed the parameter.